src/Symfony/Component/DependencyInjection/Loader/XmlFileLoader.php

```diff
@@ -317,7 +317,7 @@ private function parseDefinition(\DOMElement $service, $file, array $defaults)
                     $class = $factory->hasAttribute('class') ? $factory->getAttribute('class') : null;
                 }
 
-                $definition->setFactory([$class, $factory->getAttribute('method')]);
+                $definition->setFactory([$class, $factory->hasAttribute('method') ? $factory->getAttribute('method') : '__invoke']);
```

Comment:
IMO, I think you should test something like that : 
`!$factory->getAttribute('method') ? '__invoke' : $factory->getAttribute('method')` 

As the hasAttribute, return true if the attribute exist but set to null or empty string.
(my suggestion is used in another place in the XmlFileLoader)

Reply:
It should then be `$factory->getAttribute('method') ?: '__invoke'`

Reply:
that's right

Reply:
That is a viable alternative; it doesn't really matter which one is used from a practical standpoint as, in this context, a method attribute without a value is a userland configuration error.
Personally I prefer using `hasAttribute()` since a method attribute without a value (to me, at least) is still setting the method attribute.

But my preference doesn't matter, it'll be up to people on the DependencyInjection mergers team to decide on this I guess.

Reply:
(sorry, I didn't bother refreshing the page to see new comments! I'll update the PR now)